Split and rewrite cluster getting-started into "Access to the management cluster" and "Creating a workload cluster and installing applications", combining instructions for both product families and all providers (#1836)
giantswarm/docs: src/content/advanced/private-clusters/index.md

---
title: Private clusters

> description: By default, the Kubernetes API endpoint is public and the cluster has internet access. This article shows which products offer private cluster features.

Reviewer: ```suggestion
description: Learn how to manage private clusters in Giant Swarm. A private cluster lets you limit the Kubernetes API access and at the same time control egress traffic of your workload using a proxy.
```
I would not assume a default since we might change in future to `private` as default cluster type

weight: 35
menu:
  main:
    parent: advanced
user_questions:
- How do I make a workload cluster private?
owner:
  - https://github.com/orgs/giantswarm/teams/team-phoenix
last_review_date: 2023-06-13
---

> By default, the Kubernetes API endpoint is public and the cluster has internet access. You can skip this article unless you plan on creating clusters with strictly-limited networking.

Reviewer: ```suggestion
By default, the Kubernetes API endpoint is public and the cluster workloads have internet access. You can skip this article unless you plan on creating clusters with strictly-limited networking.
```

The following products offer private cluster features:

> - CAPA (AWS EC2)
> - CAPZ (Azure VMs)

Reviewer: ```suggestion
- CAPA ([Cluster API](https://docs.giantswarm.io/platform-overview/cluster-management/cluster-api/architecture/) implementation of AWS EC2)
- CAPZ ([Cluster API](https://docs.giantswarm.io/platform-overview/cluster-management/cluster-api/architecture/) implementation of Azure VMs)
```

Reviewer: if we are persistent with the name then plus one. Maybe we can do this as a [glossary](https://intranet.giantswarm.io/docs/glossary/#cluster-api) term and may rely on [shortcodes](https://github.com/giantswarm/docs/tree/main/src/layouts/shortcodes) so we can easily change the term afterwards. My impression is the naming is not yet solid. A PR for unification to the Glosary asking POs for review can help :)

Private clusters have special requirements:

- Since a private workload cluster still has to be reachable by the management cluster, there must be networking peering between the two. This means that the IP ranges must not overlap.
- Access to the Kubernetes API can be restricted to an internal network. It should then be reachable both by you as customer, and by Giant Swarm for support purposes. For example, a VPN can be used.

> - An HTTP proxy can be used to restrict which exact domains can be accessed at all. In this case, precise configuration is important or else the new cluster will not be able to fetch container images or reach other important resources such as internal communication to the Kubernetes API.

Reviewer: ```suggestion
- An HTTP proxy can be used to restrict which exact domains can be accessed at all. In this case, a precise configuration is essential or else the new cluster will not be able to fetch container images or reach other important resources such as internal communication to the Kubernetes API.
```

To get everything set up correctly, please get in contact with Giant Swarm. We will help creating and supporting your desired private cluster configuration within your network architecture.

> At the moment, we have these – partially still hidden – [`kubectl gs template cluster` command line options]({{< relref "/use-the-api/kubectl-gs/template-cluster" >}}) to configure access to/from the cluster:

Reviewer: I would avoid the phrase `partially still hidden` or I would add an explanation why

NAMESPACE           CLUSTER-NAME   HOST-CIDR         CONTROLPLANE-CIDR   WORKERS-CIDR      APISERVER-TYPE
org-giantswarm      glippy         [10.223.0.0/24]   [10.223.0.128/26]   [10.223.0.0/25]   Public
org-multi-project   jrp45          [10.0.0.0/16]     [10.0.0.0/20]       [10.0.16.0/20]    Public
org-observability   ytc82          [10.223.2.0/24]   [10.223.2.128/27]   [10.223.2.0/25]   Internal
```

In the example output, the next usable CIDR range is `10.223.3.0/24` and therefore the cluster configuration will look like:

```yaml
# filename: cluster-private-config.yaml
metadata:
  name: bzm29
  organization: observability
providerSpecific:
  location: westeurope
  subscriptionId: 6b1f6e4a-6d0e-4aa4-9a5a-fbaca65a4711
connectivity:
  network:
    controlPlane:
      cidr: 10.223.3.128/26
    hostCidr: 10.223.3.0/24
    mode: private
    podCidr: 192.168.0.0/16
    serviceCidr: 172.31.0.0/16
    workers:
      cidr: 10.223.3.0/25

> ```

Reviewer: What about short summary how control plane CIDR and worker CIDR is calculated?

This will create a workload cluster in the subscription `6b1f6e4a-6d0e-4aa4-9a5a-fbaca65a4711` in Azure region `westeurope`.

As all default apps are bundled in `default-apps-azure`, its basic configuration must be also given:

```yaml
# filename: default-apps-azure-config.yaml
clusterName: "bzm29"
organization: "observability"
```

The cluster template can now be generated by running:

```sh
kubectl gs template cluster --provider capz --cluster-config cluster-private-config.yaml --default-app-config default-apps-azure-config.yaml --output cluster.yaml
```

{{< /tab >}}

> {{< /tabs >}}

Reviewer: Do we have an example for AWS?
